File: src/main/java/mediathek/tool/GermanStringSorter.java
package mediathek.tool;

import org.jetbrains.annotations.NotNull;

import java.text.Collator;
import java.util.Comparator;
import java.util.Locale;

public class GermanStringSorter implements Comparator<String> {

    private static final Collator collator = Collator.getInstance(Locale.GERMANY);

    private GermanStringSorter() {
        super();
        // ignore lower/upper case, but accept special characters in localised alphabetical order
        collator.setStrength(Collator.SECONDARY);
    }

    public static GermanStringSorter getInstance() {
        return LazyHolder.INSTANCE;
    }

    @Override
    public int compare(@NotNull String o1, @NotNull String o2) {
        return collator.compare(o1, o2);
    }

    private static class LazyHolder {
        static final GermanStringSorter INSTANCE = new GermanStringSorter();
    }
}

File: src/main/kotlin/mediathek/tool/GermanStringSorter.kt
package mediathek.tool

import java.text.Collator
import java.util.*

object GermanStringSorter : Comparator<String> {
    private val collator: Collator = Collator.getInstance(Locale.GERMANY).apply {
        // ignore lower/upper case, but accept special characters in localised alphabetical order
        strength = Collator.SECONDARY
    }

    override fun compare(o1: String, o2: String): Int = collator.compare(o1, o2)
}
